v2/internal/testcommon/samples_tester_test.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package testcommon
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/conditions"
+	"github.com/google/uuid"
+)
+
+type sampleResource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Reference         genruntime.ResourceReference
+	SubscriptionID    string
+	TenantID          string
+}
+
+var _ genruntime.ARMMetaObject = &sampleResource{}
+
+func Test_SamplesTester_UpdatesSubscriptionReferences(t *testing.T) {
+	t.Parallel()
+
+	g := NewGomegaWithT(t)
+
+	tester := &SamplesTester{
+		azureSubscription: uuid.New().String(),
+	}
+
+	sample := &sampleResource{
+		Reference: genruntime.CreateResourceReferenceFromARMID("subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg/providers/Microsoft.Compute/virtualMachines/vm"),
+	}
+
+	_, err := tester.updateFieldsForTest(sample)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	g.Expect(sample.Reference.ARMID).To(ContainSubstring(tester.azureSubscription))
+}
+
+func Test_SamplesTester_UpdatesSubscriptionID(t *testing.T) {
+	t.Parallel()
+
+	g := NewGomegaWithT(t)
+
+	tester := &SamplesTester{
+		azureSubscription: uuid.New().String(),
+	}
+
+	sample := &sampleResource{
+		SubscriptionID: emptyGuid,
+	}
+
+	_, err := tester.updateFieldsForTest(sample)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	g.Expect(sample.SubscriptionID).To(Equal(tester.azureSubscription))
+}
+
+func Test_SamplesTester_UpdatesTenantID(t *testing.T) {
```

Comment:
Curious: Is this logic actually needed now with the changes you have made to the samples? Now that tenantID is being sourced from a configmap, are there other fields named `tenantID` that actually need to be replaced?

Or is this future-proofing?

Is the correct future-proofing to always expose a `tenantIDFromConfig` and `subscriptionIDFromConfig` or equivalent so that users can do what we're now doing in our samples?

Reply:
I didn't convert all the `tenantID` to `tenantIDFromConfig` - adding this support makes those cases work.